src/ChunkData.cpp

```diff
@@ -153,10 +153,10 @@ BLOCKTYPE cChunkData::GetBlock(int a_X, int a_Y, int a_Z) const
 	ASSERT((a_X >= 0) && (a_X < cChunkDef::Width));
 	ASSERT((a_Y >= 0) && (a_Y < cChunkDef::Height));
 	ASSERT((a_Z >= 0) && (a_Z < cChunkDef::Width));
-	int Section = a_Y / SectionHeight;
+	int Section = static_cast<int>(static_cast<UInt32>(a_Y) / SectionHeight);
```

Comment:
These could all have been avoided if `SectionHeight` was changed into an `int`. A much simpler solution, in my opinion.